### DISCUSSION TITLE
How to two-way bind a SourceList or SourceCache to View?

### DISCUSSION THREAD

**Discussion by LJN-hzleaper** [AvaloniaUI/Avalonia#10574] [General]
I know how to use a SourceCache to generate a ReadOnlyObservableCollection and bind view to it, but I don't know how to change a SourceCache according to user's input. In other words, how to two-way bind a SourceCache?

**Comment by timunie** [comment]
> In other words, how to two-way bind a SourceCache? 

Normally, you don't swap out the entire collection. The ObservableCollection will take care of adding or removing items. Normally the user input goes into each item. The items itself need to implement INotifyPropertyChanged in order to bind thier properites two-way. 

Example: 

```
People        (ObservableCollection)
  ⊦ Person 1  (INotifyPropertyChanged)
  |  ⌞ Name   (you can two-way bind to this)
  |  ⌞ Age    (you can two-way bind to this)
  ⊦ Person 2  (INotifyPropertyChanged)
     ⌞ Name   (you can two-way bind to this)
     ⌞ Age    (you can two-way bind to this)
```

If that is not your issue, I need more information.

**Reply by LJN-hzleaper** [reply]
Do you mean I should use ObservableCollection instead of SourceCache?

**Reply by timunie** [reply]
No SourceCache produces a ReadOnlyObservableColelction, which is totally fine. The item needs to be changed I guess. But for more support, please show me what exactly is not working. 

- How does your item look?
- How does your XAML look?
